src/SimpleSAML/Session.php

```diff
@@ -964,6 +965,10 @@ public function getData(string $type, ?string $id): mixed
             return null;
         }
 
+        if (!$allowExpired && $this->dataStore[$type][$id]['expires'] < time()) {
```

Comment:
This should also take into account the case where expiration was set to string `self::DATA_TIMEOUT_SESSION_END`

Reply:
done